### PR TITLE
Change build_ctc_topo to put the blank symbol into `labels`.

### PR DESCRIPTION
**Pull request by csukuangfj**

### CODE REVIEW

**Comment by danpovey** [comment]
LGTM, perhaps one of the other guys can merge if they think it's OK.

File: k2/python/tests/ctc_gradients_test.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ def build_ctc_topo(tokens: List[int]) -> k2.Fsa:
     symbols to a single output symbol.
 
     Caution:
-      The resulting topo is an FST. Epsilons are on the left
-      side (i.e., ilabels) and tokens are on the right side (i.e., olabels)
+      The resulting topo is an FST. Epsilons are on the right
+      side (i.e., olabels) and tokens are on the left side (i.e., olabels)
```
Comment (qindazhu) [inline k2/python/tests/ctc_gradients_test.py:27]: left side is ilabels?

**Comment by danpovey** [comment]
Actually the whole comment is not well worded IMO.
Maybe  "A token which appears once on the right side (i.e. olabels) may
appear multiple times on the
    left side (ilabels), possibly with epsilons in between."
would be better.

**Comment by danpovey** [comment]
.. or maybe "blanks in between"

**Comment by qindazhu** [comment]
I would suggest we clarify `blank` and `epsilon` in the documentation as well so readers will not be confused.

File: k2/python/tests/ctc_gradients_test.py (hunk 1 above)
Comment (csukuangfj) [reply k2/python/tests/ctc_gradients_test.py]: fixed, thanks.

**Comment by csukuangfj** [comment]
> Actually the whole comment is not well worded IMO.

Thanks, changed as you suggested.

**Comment by csukuangfj** [comment]
> I would suggest we clarify blank and epsilon in the documentation as well so readers will not be confused.

Do you have any idea of how to make it clearer?

**Comment by yaguanghu** [comment]
Maybe keep `epsilon` in ilables is a solution? Then we can use `1` or `num_of_tokens + 1` as `blank` index.

**Comment by qindazhu** [comment]
> Maybe keep `epsilon` in ilables is a solution? Then we can use `1` or `num_of_tokens + 1` as `blank` index.

I think we have used `blank == 0` in C++ intersect code, so defining it as other values may be not a good idea. `when 0 appears on left side, it represents blank` is fine with me.

File: k2/python/tests/ctc_gradients_test.py [hunk 2]
```diff
@@ -36,20 +37,43 @@ def build_ctc_topo(tokens: List[int]) -> k2.Fsa:
...
             if i == j:
-                rules += f'{i} {i} 0 {tokens[i]} 0.0\n'
+                arcs += f'{i} {i} {tokens[i]} 0 0.0\n'
             else:
-                rules += f'{i} {j} {tokens[j]} {tokens[j]} 0.0\n'
-        rules += f'{i} {final_state} -1 -1 0.0\n'
-    rules += f'{final_state}'
-    ans = k2.Fsa.from_str(rules)
-    ans = k2.arc_sort(ans)
+                # we assume that eps has the same ID, i.e., 0, with
```
Comment (qindazhu) [inline k2/python/tests/ctc_gradients_test.py:46]: I suppose we can delete these comments with above `CAUTION`?

Comment (csukuangfj) [reply k2/python/tests/ctc_gradients_test.py:46]: fixed.

**Comment by csukuangfj** [comment]
Merging.

**Pull request merged**